--- mythtv/libs/libmythtv/recorders/dvbchannel.cpp_backup	2015-01-06 10:42:14.349222058 +0000
+++ mythtv/libs/libmythtv/recorders/dvbchannel.cpp	2015-01-06 10:44:01.592091685 +0000
@@ -548,7 +548,7 @@
     if (!cmdseq)
         return NULL;
 
-    cmdseq->props = (struct dtv_property*) calloc(11, sizeof(*(cmdseq->props)));
+    cmdseq->props = (struct dtv_property*) calloc(12, sizeof(*(cmdseq->props)));
     if (!(cmdseq->props))
     {
         free(cmdseq);
@@ -609,6 +609,22 @@
         cmdseq->props[c++].u.data = PILOT_AUTO;
         cmdseq->props[c].cmd      = DTV_ROLLOFF;
         cmdseq->props[c++].u.data = tuning.rolloff;
+
+        LOG(VB_CHANNEL, LOG_INFO, QString("SETTING MIS! %1").arg(tuning.frequency));
+        cmdseq->props[c].cmd = DTV_STREAM_ID;
+	if (tuning.frequency % 10 == 0) {
+            cmdseq->props[c++].u.data = 0;
+        } else {
+            int stream_id, pls_code, pls_mode;
+	    stream_id = tuning.frequency % 10;
+            pls_code = 121212;
+            pls_mode = 1;
+            cmdseq->props[c].cmd = DTV_STREAM_ID;
+            cmdseq->props[c++].u.data = 
+		((pls_mode & 0x3) << 26) |
+    	        ((pls_code & 0x3ffff) << 8) |
+    		(stream_id & 0xff);
+    	}
     }
     else if (tuning.mod_sys == DTVModulationSystem::kModulationSystem_DVBS)
     {
